etna/transforms/encoders/mean_segment_encoder.py

```diff
@@ -1,24 +1,35 @@
-import reprlib
-from typing import Dict
 from typing import List
-from typing import Optional
 
 import numpy as np
 import pandas as pd
 
 from etna.transforms import IrreversibleTransform
-from etna.transforms.math.statistics import MeanTransform
+from etna.transforms.encoders.mean_encoder import MeanEncoderTransform
 
 
 class MeanSegmentEncoderTransform(IrreversibleTransform):
     """Makes expanding mean target encoding of the segment. Creates column 'segment_mean'."""
 
     idx = pd.IndexSlice
+    segment_column = "segment_column"
+    out_column = "segment_mean"
 
     def __init__(self):
         super().__init__(required_features=["target"])
-        self.mean_encoder = MeanTransform(in_column="target", window=-1, out_column="segment_mean")
-        self.global_means: Optional[Dict[str, float]] = None
+        self.mean_encoder = MeanEncoderTransform(
+            in_column=self.segment_column, mode="per-segment", out_column=self.out_column, smoothing=0
```

Comment:
Don't we want to add `smoothing` parameter?

Reply:
Where?

Reply:
I'm suggesting that we could add `smoothing` parameter into `MeanSegmentEncoderTransform`.

Reply:
See my previous comment